packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -362,8 +362,12 @@ class ReactSixteenAdapter extends EnzymeAdapter {
         return instance ? toTree(instance._reactInternalFiber).rendered : null;
       },
       simulateError(nodeHierarchy, rootNode, error) {
-        const { instance: catchingInstance } = nodeHierarchy
-          .find(x => x.instance && x.instance.componentDidCatch) || {};
+        const isErrorBoundary = ({ instance: elInstance, type }) => (
+          (elInstance && elInstance.componentDidCatch) || (type && type.getDerivedStateFromError)
```

Comment:
probably only in 16.6+ should gDSFE's presence matter?
```suggestion
          (elInstance && elInstance.componentDidCatch) || (is166 && type && type.getDerivedStateFromError)
```